### PR TITLE
fix two minor issues

### PR DESCRIPTION
**Pull request by tabokie**
- Sync file after truncating tail corruption.
- Fix a header decoding check.

### CODE REVIEW

File: src/log_batch.rs [hunk 1]
```diff
@@ -891,9 +891,10 @@ impl LogBatch {
             )));
         }
 
-        let len = codec::decode_u64(buf)? as usize;
+        let len_and_type = codec::decode_u64(buf)? as usize;
+        let compression_type = CompressionType::from_u8(len_and_type as u8)?;
+        let len = len_and_type >> 8;
```
Comment (BusyJay) [inline src/log_batch.rs:896]: Any tests?

Comment (tabokie) [reply src/log_batch.rs:896]: added

File: src/log_batch.rs [hunk 2]
```diff
@@ -1531,6 +1531,49 @@ mod tests {
...
+            .add_entries::<Entry>(region_id, &generate_entries(1, 11, Some(&data)))
+            .unwrap();
+        batch
+            .put(region_id, b"key".to_vec(), b"value".to_vec())
+            .unwrap();
+        // enable compression so that len_and_type > len.
+        batch.finish_populate(1).unwrap();
+        let file_context = LogFileContext::new(FileId::dummy(LogQueue::Append), Version::default());
+        batch.prepare_write(&file_context).unwrap();
+        let encoded = batch.encoded_bytes();
```
Comment (BusyJay) [inline src/log_batch.rs:1549]: Still missing a valid decode so we can verify it won't panic anymore.

**Review by BusyJay — APPROVED** [review]

File: src/log_batch.rs (hunk 2 above)
Comment (tabokie) [reply src/log_batch.rs:1549]: There're plenty of tests above testing the valid cases.

**Pull request merged**